presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1574,6 +1576,13 @@ public OptionalLong metadataDelete(ConnectorSession session, ConnectorTableHandl
             throw new TableNotFoundException(handle.getSchemaTableName());
         }
 
+        List<ColumnHandle> partitionColumns = layoutHandle.getPartitionColumns();
+        if (!layoutHandle.getEffectivePredicate().getDomains()
+                .map(domains -> Maps.filterKeys(domains, not(Predicates.in(partitionColumns))))
```

Comment:
To me this would either be easier to read if it were written imperatively, or if we consistently statically imported things:

`.map(domains -> filterKeys(domains, not(in(partitionColumns))))`